junit-jupiter-params/src/main/java/org/junit/jupiter/params/aggregator/ArgumentsAccessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.params.aggregator;
+
+import java.util.List;
+
+public interface ArgumentsAccessor {
+
+	Object[] toArray();
+
+	List<Object> toList();
+
+	int size();
+
+	Object get(int index);
+
+	<T> T get(Class<T> clazz, int index);
+
+	Character getChar(int index);
+
+	Boolean getBoolean(int index);
```

Comment:
No reason at all, just assumed this was your decision for visual consistency. Correcting now

Reply:
It's not about visual consistency. Rather, it's about runtime semantics.

If it would ever make sense for the returned value to be `null`, we would then have to use wrapper types.

Otherwise, if the returned value can never be `null`, we should then use primitive types where possible.

In other words, I was basically asking if you had come up with a scenario where returning `null` would make sense.

Make sense? 😉

Reply:
Well the converter does "convert" null to any reference type ,as in, does not throw an exception. So if there is any possibility of a provided argument being null, aggregators could batch convert arguments without crashing. Waiting for your instructions on this

Reply:
Well, it's possible for a user to specify a `null` value in the source for a given index and at other times provide a non-null value for the same index.

So I guess your original proposal is correct to allow `null` converted values.

In other words, I'm thinking we should stick with wrappers instead of primitive types for the greatest flexibility in the API. Consumers of the API can then decide on their own what to do about required vs. optional values.